### PR TITLE
[API] Dashboard list doc

### PR DESCRIPTION
**Pull request by MLaureB**
### What does this PR do?

Adds API documentation for a new resource `Dashboard List` with 9 new endpoints: 

API endpoints for managing dashboard lists:

- Get all dashboard lists: `GET dashboard/lists/manual`

- Get dashboard list: `GET dashboard/lists/manual/{dashboard_list_id}`

- Create dashboard list: `POST dashboard/lists/manual`

- Update dashboard list: `PUT dashboard/lists/manual/{dashboard_list_id}`

- Delete dashboard list: `DELETE dashboard/lists/manual/{dashboard_list_id}`

- Get dashboards for dashboard list: `GET dashboard/lists/manual/{dashboard_list_id}/dashboards`

- Add dashboards to dashboard list: `POST dashboard/lists/manual/{dashboard_list_id}/dashboards`

- Update dashboards of dashboard list: `PUT dashboard/lists/manual/{dashboard_list_id}/dashboards`

- Delete dashboards from dashboard list: `DELETE dashboard/lists/manual/{dashboard_list_id}/dashboards`

### Motivation

We want to expose to our customers the different endpoints for managing dashboard lists via the API.

### Preview link

https://docs-staging.datadoghq.com/marielaure.bardonnet/dashboard_list_doc/api/?lang=python#dashboard-lists

### Additional Notes

In the app, the different lists of dashboards can be accessed here: 
https://app.datadoghq.com/dashboard/lists

### CODE REVIEW

File: content/api/dashboard_lists/code_snippets/result.api-dashboard-list-get-all.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
+[
+    "200",
+    {
+        "dashboard_lists" => [
+            {
+                "is_favorite" => true,
+                "name" => "My Dashboard List B",
+                "dashboard_count" => 22,
+                "author" => {
+                    "handle" => "author.name@datadoghq.com",
```
Comment (jyee) [inline content/api/dashboard_lists/code_snippets/result.api-dashboard-list-get-all.rb]: Let's keep this consistent with the other api doc example code and use test1@datadoghq.com and test2@datadoghq.com

File: content/api/dashboard_lists/code_snippets/result.api-dashboard-list-get-items.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            'is_favorite': True,
+            'is_read_only': True,
+            'type': 'integration_timeboard',
+            'id': 17,
+            'icon': '/static/v/34.254868/images/saas_logos/small/postgres.png'
+        },
+        {
+            'is_shared': False,
+            'author': {
+                'handle': 'leo@datadoghq.com',
```
Comment (jyee) [inline content/api/dashboard_lists/code_snippets/result.api-dashboard-list-get-items.py]: Similar to other comments, let's use test1 and test2@datadoghq.com. Never a real email address.

File: content/api/dashboard_lists/dashboard_lists_get_all.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
+---
+title: Get all Dashboard Lists
+type: apicontent
+order: 8.2
+external_redirect: /api/#get-all-dashboard-lists
+---
+
+## Get all Dashboard Lists
+
+Fetch all of your existing dashboard list’s definitions.
```
Comment (jyee) [inline content/api/dashboard_lists/dashboard_lists_get_all.md]: I'm not sure what the Datadog standard is for plural possessive. I think this probably should be either `lists' definitions` or `lists's definitions`. Probably worth tagging in @johnaxel to comment.

File: content/api/dashboard_lists/dashboard_lists_update_items.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
+---
+title: Update Items of a Dashboard List
+type: apicontent
+order: 8.8
+external_redirect: /api/#update-items-of-a-dashboard-list
+---
+
+## Update Items of a Dashboard List
+
+Override dashboards of an existing dashboard list.
```
Comment (jyee) [inline content/api/dashboard_lists/dashboard_lists_update_items.md]: s/Override/Update/

File: content/api/downtimes/downtimes_schedule.md [hunk 5]
```diff
@@ -1,31 +1,30 @@
...
 external_redirect: /api/#schedule-monitor-downtime
 ---
 
 ## Schedule monitor downtime
 
 ##### ARGUMENTS
 
-
-* **`scope`** [*required*]:  
+*   **`scope`** [*required*]:
```
Comment (jyee) [inline content/api/downtimes/downtimes_schedule.md]: This file has new indentations that aren't part of the monitor lists PR. Is this necessary?

File: content/api/downtimes/downtimes_update.md [hunk 6]
```diff
@@ -1,31 +1,32 @@
...
+order: 9.2
 external_redirect: /api/#update-monitor-downtime
 ---
 
 ## Update monitor downtime
+
 ##### ARGUMENTS
 
-* **`id`** [*required*]:  
+*   **`id`** [*required*]:
```
Comment (jyee) [inline content/api/downtimes/downtimes_update.md]: This file has new indentations that aren't part of the monitor lists PR. Is this necessary?

File: content/api/embeds/embeds_create.md [hunk 7]
```diff
@@ -14,23 +14,24 @@ Returns: A JSON consisting of the same elements returned by GET api/v1/graph/emb
 Note: If an embed already exists for the exact same query in a given organization, the older embed is returned instead of creating a new embed.
 
 ##### ARGUMENTS
-* **`graph_json`** [*required*]:  
+
+*   **`graph_json`** [*required*]:
```
Comment (jyee) [inline content/api/embeds/embeds_create.md]: This file has new indentations that aren't part of the monitor lists PR. Is this necessary?

File: content/api/events/events_post.md [hunk 8]
```diff
@@ -1,35 +1,37 @@
...
 ## Post an event
+
 This end point allows you to post events to the stream. Tag them, set priority and event aggregate them with other events.
 
 ##### ARGUMENTS
-* **`title`** [*required*]:  
-    The event title. *Limited to 100 characters.*  
+
+*   **`title`** [*required*]:
+    The event title. _Limited to 100 characters._
```
Comment (jyee) [inline content/api/events/events_post.md]: Is there any reason we're changing from single `*` to `_`?

File: content/api/dashboard_lists/dashboard_lists_get_all.md (hunk 3 above)
Comment (technovangelist) [reply content/api/dashboard_lists/dashboard_lists_get_all.md]: lists's is always wrong. but lists' definitions is ok. but wait. is this getting all the lists? so dashboard list definitions is better.

File: content/api/events/events_post.md (hunk 8 above)
Comment (technovangelist) [reply content/api/events/events_post.md]: yup, no reason to change this. I find it makes it more confusing to use different symbols in different places. they are both correct, but the asterisk is a bit more common.

File: content/api/dashboard_lists/dashboard_lists_get_all.md (hunk 3 above)
Comment (johnaxel) [reply content/api/dashboard_lists/dashboard_lists_get_all.md]: I agree that "dashboard list definitions" is best.

File: content/api/events/events_post.md (hunk 8 above)
Comment (MLaureB) [reply content/api/events/events_post.md]: My code editor did those changes automatically when saving the `.md`, I'll try to remove those changes.

**Pull request merged**